server/src/handlers/logstream.rs

```diff
@@ -46,21 +48,30 @@ pub async fn delete(req: HttpRequest) -> HttpResponse {
         .to_http();
     }
 
-    if let Err(e) = s3.delete_stream(&stream_name).await {
+    if let Err(e) = metadata::STREAM_INFO.delete_stream(&stream_name) {
```

Comment:
order of deletion matters here. First delete from s3 because that has highest probability to fail. Only on successfully deleting from s3, attempt local directory and then metadata

If s3 deletion fails, fail the api
if s3 delete worked and local dir delete failed, respond with 200 in api response and log on console about failed delete in local data dir.
if s3 and local dir delete worked and metadata delete failed, same as second case, api is successful and error log on console about exact failure